lib/active_merchant/billing/gateways/adyen.rb

```diff
@@ -435,17 +435,17 @@ def add_splits(post, options)
 
         splits = []
         split_data.each do |split|
-          amount = {
-            value: split['amount']['value']
-          }
-          amount[:currency] = split['amount']['currency'] if split['amount']['currency']
-
-          split_hash = {
-            amount: amount,
-            type: split['type'],
-            reference: split['reference']
-          }
-          split_hash['account'] = split['account'] if split['account']
+          if split['amount']
+            amount = {}
+            amount[:value] = split['amount']['value'] if split['amount']['value']
+            amount[:currency] = split['amount']['currency'] if split['amount']['currency']
+          end
+
+          split_hash = {}
+          split_hash[:amount] = amount unless split['type'] == 'PaymentFee' || split['type'] == 'AdyenFees' || amount.nil?
```

Comment:
Instead of specifying the types to omit, could we just check for the presence of `amount` and make sure that it's not an empty hash?

Reply:
This is a simpler way to omit the amount object, I think I originally wrote it to omit just in case they send the amount alng with the type being 'PaymentFee' or 'AdyenFees'. Almost like an extra guard but i guess they can refer to the Adyens documentation